### PR TITLE
Ruelvla 4 - Mock Badges Endpoint - GET

### PR DESCRIPTION
**Pull request by vmikii**

### CODE REVIEW

File: rueppellii-lvlup-asp.net-core.sln [hunk 1]
```diff
@@ -4,8 +4,11 @@ Microsoft Visual Studio Solution File, Format Version 12.00
 VisualStudioVersion = 15.0.28307.168
 MinimumVisualStudioVersion = 10.0.40219.1
 Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "rueppellii-lvlup-asp.net-core", "rueppellii-lvlup-asp.net-core\rueppellii-lvlup-asp.net-core.csproj", "{9B1ECEFB-5787-4D28-BAFE-EB58EEE1B5EB}"
+<<<<<<< HEAD
+=======
 EndProject
 Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "rueppellii-lvlup-asp.net-core.IntegrationTests", "rueppellii-lvlup-asp.net-core.IntegrationTests\rueppellii-lvlup-asp.net-core.IntegrationTests.csproj", "{B57A7389-0738-4007-8995-685FD70D5E46}"
+>>>>>>> 61fa3c58b95da3c4afabd5a5042ede5f306a3ca7
```
Comment (szirmi) [inline rueppellii-lvlup-asp.net-core.sln]: resolve the conflict please

File: rueppellii-lvlup-asp.net-core/Controllers/BadgesController.cs [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Threading.Tasks;
+
+namespace rueppellii_lvlup_asp.net_core.Controllers
+{
+  [ApiController]
+  [Route("[controller]")]
```
Comment (szirmi) [inline rueppellii-lvlup-asp.net-core/Controllers/BadgesController.cs]: this is not equal to the route specified
http://jira.greenfox.academy/projects/RUELVLA/issues/RUELVLA-4

File: rueppellii-lvlup-asp.net-core/Controllers/BadgesController.cs [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+  public class BadgesController : Controller
+  {
+    [HttpGet]
+    public IActionResult ListBadges()
+    {
+      var badges = "{ \"badges\": [ { \"name\": \"Process improver\", \"level\": \"2\"}, { \"name\": \"English speaker\", \"level\": \"1\"}, { \"name\": \"Feedback giver\", \"level\": \"1\"} ] }";
```
Comment (szirmi) [inline rueppellii-lvlup-asp.net-core/Controllers/BadgesController.cs]: in general that is better solution if you create an array of Badge objects and then stringify it

File: rueppellii-lvlup-asp.net-core.IntegrationTests/Scenarios/BadgesControllerTests.cs [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+      var request = new HttpRequestMessage(HttpMethod.Get, "/badges");
+      request.Headers.Add("usertokenauth", "invalid");
+      var response = await testContext.Client.SendAsync(request);
+      Assert.Equal(HttpStatusCode.Unauthorized, response.StatusCode);
+    }
+
+    [Fact]
+    public async Task ListBadges_Should_ReturnBadges()
+    {
+      string badges = "{ \"badges\": [ { \"name\": \"Process improver\", \"level\": \"2\"}, { \"name\": \"English speaker\", \"level\": \"1\"}, { \"name\": \"Feedback giver\", \"level\": \"1\"} ] }";
```
Comment (szirmi) [inline rueppellii-lvlup-asp.net-core.IntegrationTests/Scenarios/BadgesControllerTests.cs]: in general that is better solution if you create an array of Badge objects and then stringify it

**Review by szirmi — CHANGES_REQUESTED** [review]

**Comment by vmikii** [comment]
Requested changes done.

**Review by szirmi — APPROVED** [review]

**Pull request merged**